runtime/runtime/src/lib.rs

```diff
@@ -1070,6 +1072,48 @@ impl Runtime {
         Ok(())
     }
 
+    // In test runs reads and writes here used 442 TGas. Added 10% to account for possible bigger
+    // state
+    #[cfg(feature = "protocol_feature_fix_storage_usage")]
+    const GAS_USED_FOR_STORAGE_USAGE_DELTA_MIGRATION: Gas = 490_000_000_000_000;
+
+    pub fn apply_migrations(
+        &self,
+        state_update: &mut TrieUpdate,
+        apply_state: &ApplyState,
+    ) -> Result<Gas, StorageError> {
+        #[cfg(feature = "protocol_feature_fix_storage_usage")]
+        let mut gas_used: Gas = 0;
+        #[cfg(not(feature = "protocol_feature_fix_storage_usage"))]
+        let gas_used: Gas = 0;
+        #[cfg(feature = "protocol_feature_fix_storage_usage")]
+        if ProtocolFeature::FixStorageUsage.protocol_version()
+            == apply_state.current_protocol_version
```

Comment:
should we move this logic to the construction of `ApplyState`?

Reply:
I do not think so, tbh, but can do that if you insist

Reply:
After some consideration I agree

Reply:
After some more consideration I again think this is more appropriate place to select exact migration. On the other hand passing non-empty MigrationData only on new version blocks seems reasonable

Reply:
See my new implementation

Reply:
@EgorKulikov could you explain why you think it is better to put this logic here? My reasoning is as follows:
* If we move this logic to when `ApplyState` is constructed, all the logic for determining whether migration data should be populated will be in one place and it is easier to read and reason about.
* You don't need to have the `unreachable` branch here -- `None` would mean that no migration is needed.

Reply:
Your approach may well be better if we only have one item in MigrationData, but how do you think we should deal with multiple such elements?

Reply:
Refactored apply_migrations for more clarity